mithril-common/src/store/adapter/sqlite_adapter.rs

```diff
@@ -108,11 +108,24 @@ where
     }
 
     fn fetch_maybe_one_value(&self, mut statement: Statement) -> Result<Option<V>> {
-        if State::Done
-            == statement
-                .next()
-                .map_err(|e| AdapterError::ParsingDataError(e.into()))?
-        {
+        let mut retries = Some(3_u32);
+        let mut result = statement.next();
+
+        while result.is_err() {
+            // database is probably locked
+            // wait and retry strategy
+            retries = retries.filter(|v| v > &1).map(|v| v - 1);
+
+            if retries.is_none() {
+                return Err(result
+                    .map_err(|e| AdapterError::ParsingDataError(e.into()))
+                    .unwrap_err());
+            }
+            sleep(Duration::from_millis(50));
```

Comment:
You could maybe declare a variable for wait duration? 
Also, it would be nice to have exponential back-off retry strategy, WDYT?

BTW, is there any way to test the retries?

Reply:
- Yes for the constant 👍🏻  
 - IMHO it is unnecessary to apply complex strategies here since simple ones work
 - the best way to test is to make the end to end test to work flawlessly (which seems to be the case with that patch)